pkg/pkg.mk

```diff
@@ -32,7 +32,9 @@ endif
 git-ensure-version: $(PKG_BUILDDIR)/.git-downloaded
 	if [ $(shell git -C $(PKG_BUILDDIR) rev-parse HEAD) != $(PKG_VERSION) ] ; then \
 		git -C $(PKG_BUILDDIR) clean -xdff ; \
-		git -C $(PKG_BUILDDIR) fetch "$(PKG_URL)" "$(PKG_VERSION)" ; \
+		if git cat-file -e -t '$(PKG_VERSION)^{commit}' ; then \
```

Comment:
On `ubuntu` bionic, git does not like `-e -t`.
`error: switch `t' : incompatible with -e`.

I would like to maybe put it outside in a "git_check_version_is_a_commit" so document that it detects branches.

Reply:
```
git cat-file -e 2019.04^{commit} && echo commit
commit
```
```
git cat-file -e 2019.04-branch^{commit}
fatal: Not a valid object name 2019.04-branch^{commit}
```

Also works with a hash as expected.